backend/contract-service/src/main/java/org/egov/works/service/NotificationService.java

```diff
@@ -0,0 +1,336 @@
+package org.egov.works.service;
+
+import com.jayway.jsonpath.JsonPath;
+import digit.models.coremodels.RequestInfoWrapper;
+import digit.models.coremodels.SMSRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.works.config.ContractServiceConfiguration;
+import org.egov.works.kafka.Producer;
+import org.egov.works.repository.ServiceRequestRepository;
+import org.egov.works.util.*;
+import org.egov.works.web.models.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class NotificationService {
+
+    @Autowired
+    private Producer producer;
+
+    @Autowired
+    private ServiceRequestRepository repository;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private ContractServiceConfiguration config;
+
+    @Autowired
+    private HRMSUtils hrmsUtils;
+
+    @Autowired
+    private EstimateServiceUtil estimateServiceUtil;
+
+    @Autowired
+    private ProjectServiceUtil projectServiceUtil;
+
+    @Autowired
+    private LocationServiceUtil locationServiceUtil;
+
+    @Autowired
+    private OrganisationServiceUtil organisationServiceUtil;
+
+
+    /**
+     * Sends notification by putting the sms content onto the core-sms topic
+     *
+     * @param request
+     */
+    public void sendNotification(ContractRequest request) {
+        Workflow workflow = request.getWorkflow();
+
```

Comment:
pushNotificationToCreatorForRejectAction, pushNotificationToCreatorForApproveAction and pushNotificationToCBOForApproveAction can not be merged into single method?

Reply:
I started with a single method. But CK suggests that message templates are bound to change and we need to adapt the entire code later to accomodate those changes. Hence, keeping the code flexible to accomodate these changes